src/CardForm/CardForm.js

```diff
@@ -94,10 +96,11 @@ class CardForm extends Component {
   };
 
   handleChange = (name, value) => {
+    const { onChange } = this.props;
     const { values: previousValues } = this.state;
     const values = { ...previousValues, [name]: value };
 
-    this.setState({ values });
+    this.setState({ values }, onChange(name, value));
```

Comment:
Does it makes sense to execute the on change when the state is totally changed? Not sure if we are wining anything executing it as a callback fn.

Reply:
I added it to notify parent components about any change in the form. We need it in OPP to know if the form is dirty or not but I'm open to other alternatives :)

Reply:
I think borja suggests just calling onChange after the setState, not as it's argument.
if the cb may rerender something within the component it might be ok to leave it as it is? not sure